src/components/MCQBlock.tsx

```diff
@@ -82,19 +91,31 @@ export default function MCQBlock({
   }
 
   const handleSubmission = () => {
-    setSubmitted(true)
-    setAttempts(attempts - 1)
+    if (selectedAnswers.length != correctOptions.length) {
+      setWarning(true)
+    } else {
+      setWarning(false)
+      setSubmitted(true)
+      setAttempts(attempts - 1)
```

Comment:
When the new state depends on the previous state, the recommended way of updating the state is using [functional updates](https://reactjs.org/docs/hooks-reference.html#functional-updates).

For example, `setState(state => state + 1)`

Reply:
- Well, I think we should remove attempts completely like you said submit as many times since it makes more sense. I will update that.

- For verdict, I guess since there is no attempts, we can just show all correct answer once submit (even on first try) and incorrect if wrong options selected. 

- For Hint, I think it would be better to have it as an icon than a complete button